### PR TITLE
Shadow assoc

### PR DESCRIPTION
**Pull request by kirbyfan64**
This merges paultag's branch that fixes apply with shadowed operators and adds tests, as well as shadowing assoc.

This #724 and #647.

### CODE REVIEW

File: tests/native_tests/language.hy [hunk 1]
```diff
@@ -264,6 +264,15 @@
              "a b c d e")))
 
 
+(defn test-apply-with-shadows []
+  "NATIVE: test apply to call shadowed operators"
+  (assert (= (apply + [1 2]) 3))
+  (assert (= (apply - [1 2]) -1))
+  (setv d {})
+  (apply assoc [d 123 456])
+  (assert (= (get assoc 123) 456)))
```
Comment (paultag) [inline tests/native_tests/language.hy]: @kirbyfan64 do you mean `(get d 123)` ?

**Comment by paultag** [comment]
This also seems to break an unrelated test -- thrilling

File: tests/native_tests/language.hy (hunk 1 above)
Comment (paultag) [inline tests/native_tests/language.hy]: ping :)

**Comment by kirbyfan64** [comment]
Calling a Git-savvy user...

I can't find this branch. At all. Is there a reason a branch wouldn't appear with `git branch -va`?

**Comment by zackmdavis** [comment]
@kirbyfan64 It's definitely on [your GitHub fork](https://github.com/kirbyfan64/hy/tree/shadow_assoc); one can only imagine your local clone doesn't have that ref for whatever reason? Maybe `git remote add kirbyfan64 git@github.com:kirbyfan64/hy.git; git fetch kirbyfan64` (add your fork as a remote if it didn't exist, and fetch from it) and `git checkout shadow_assoc` (git will create a new tracking branch when there's one on a remote with a matching name)?

**Comment by kirbyfan64** [comment]
@zackmdavis I had it as a remote, but I didn't think about running `git fetch`. That did the trick. Thanks!

**Comment by paultag** [comment]
Tests are failing, re-open when this guy's ready!

**Pull request closed without merging**